finetune/base_models/bert/featurizer.py

```diff
@@ -192,32 +192,38 @@ def long_doc_bert_featurizer(
     #     initial_shape, ["initial_shape", initial_shape], summarize=-1
     # )
 
-    batch_size = initial_shape[0]
-    sequence_len = initial_shape[1]
-
     # [batch_size, sequence_length]
     X = tf.reshape(X, shape=tf.concat(([-1], initial_shape[-1:]), 0))
     X.set_shape([None, None])
-    X = tf.compat.v1.Print(X, ["X", X, tf.shape(X)], summarize=-1)
+    if config.extra_print:
+        X = tf.compat.v1.Print(X, ["X", X, tf.shape(X)], summarize=-1)
+
+    # Remove start tokens from tensors, because they will be added back per chunk
+    batch_size = initial_shape[0]
+    sequence_len = initial_shape[-1] - 1
+    X = X[:, 1:]
+    if config.extra_print:
+        X = tf.compat.v1.Print(X, ["X_remove_start", X, tf.shape(X)], summarize=-1)
 
     # Add padding to X to later break out a separate chunk dimension
     """
     FIXME Remove comments with math. Those are with these assumptions for validating
     padding / chunking logic
     * config.chunk_size = 64
     * config.batch_size_scaler = 4
-    * sequence_len = 300
+    * sequence_len = 300 - 1 = 299
     """
     # Need to effectively "scale" chunk dim to account for division by batch_size_scaler so that
     # num_chunks is divisible by batch_size_scaler. In some cases this requires padding by a lot
     # of zeros
     effective_chunk_dim = config.chunk_size * config.batch_size_scaler    # 64 * 4 = 256
-    pad_count = effective_chunk_dim - (sequence_len % effective_chunk_dim)  # 256 - (300 % 256) = 212
-    new_seq_len = sequence_len + pad_count  # 300 + 212 = 512
-    num_chunks = new_seq_len / config.chunk_size  # 512 / 64 = 8
+    pad_count = effective_chunk_dim - (sequence_len % effective_chunk_dim)  # 256 - (299 % 256) = 197
+    new_seq_len = sequence_len + pad_count  # 299 + 213 = 512
```

Comment:
Do we want to pad this up to full sequence len or sequence len - 2 to account for the extra tokens that will be added?

Reply:
I'm adding the extra tokens after I already reshape to [batch_size * batch_scaler, num_chunks / batch_scaler, chunk_size], so I don't think I need extra padding? But I might be missing something